server/app/repository/ApplicationEventRepository.java

```diff
@@ -68,6 +64,58 @@ public ImmutableList<ApplicationEventModel> getEventsOrderByCreateTimeDesc(Long
             .findList());
   }
 
+  /**
+   * Updates the ApplicationEvents and the Applications table to the latest status in a single
+   * transaction.
+   *
+   * <p>Previously in 44.sql, DB triggers were used to update the latest_status between
+   * application_events table and applications table. In 78.sql script, the team decided to drop
+   * triggers and use application code to update both tables. This method implements the same.
+   *
+   * <p>Note - Application code must change both tables at once as we want to avoid inconsistency
+   * between the tables.
+   */
+  public CompletionStage<ApplicationEventModel> insertStatusEvent(
+      ApplicationModel application,
+      Optional<AccountModel> optionalAdmin,
+      ApplicationEventDetails.StatusEvent newStatusEvent) {
+    ApplicationEventDetails details =
+        ApplicationEventDetails.builder()
+            .setEventType(ApplicationEventDetails.Type.STATUS_CHANGE)
+            .setStatusEvent(newStatusEvent)
+            .build();
+    ApplicationEventModel event = new ApplicationEventModel(application, optionalAdmin, details);
+    return supplyAsync(
+        () -> {
+          try (Transaction transaction = database.beginTransaction(TxIsolation.SERIALIZABLE)) {
+            insertSync(event);
+            // Saves the latest note on the applications table too.
+            // If the status is removed from an application, then the latest_status column needs
+            // to be set to null to indicate the application has no status and not a status with
+            // empty string.
+            if (Strings.isNullOrEmpty(newStatusEvent.statusText())) {
+              database
+                  .update(ApplicationModel.class)
+                  .set("latest_status", null)
+                  .where()
+                  .eq("id", application.id)
+                  .update();
+            } else {
+              database
+                  .update(ApplicationModel.class)
+                  .set("latest_status", newStatusEvent.statusText())
+                  .where()
+                  .eq("id", application.id)
+                  .update();
+            }
```

Comment:
(blocking) The could be simplified by getting the value `Strings.isNullOrEmpty(newStatusEvent.statusText()) ? null : newStatusEvent.statusText()` so you only have to write the update once and not sorry about having to change it in two places down the road.

Reply:
Changed to blocking

Reply:
Jolly good! will do it right now!